lib/collection/src/collection.rs

```diff
@@ -916,14 +996,41 @@ impl Collection {
             try_join_all(all_searches).await?
         };
 
+        // TODO(luis): see if we can avoid this clone by removing the Arc
```

Comment:
Want to keep the TODO for later?

Reply:
Whoops, must've reappeared when rebasing